architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,296 @@
+
+## Terms and Definitions
+
+| Term                  | Definition                                                                                                                                                                                  |
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| RP                 | Resource Provider    |
+| UCP                 | Universal Control Plane for Radius   |
+| DE | Deployment Engine |
+
+## System Description
+
+Applications RP is a Radius control-plane microservice that acts as resource provider for application and its resources. When users deploy an application using Radius, Applications RP manages the lifecycle of the created resources on the user's behalf. 
+
+- Deploying an application may launch the user application's code on the same cluster as Radius, or a different cluster. 
+
+- Deploying application may create resources in the cloud using Recipes. These *recipes* are bicep or terraform code, and are responsible for creating   application infrastructure components like databases.
+
+- As a result, Applications RP has access to the user's cloud credentials and can manage the user's cloud resources.
+
+Resource providers (including Applications RP) communicate over HTTP and manage the lifecycle of resources. See the [architecture documentation](https://docs.radapp.io/concepts/technical/architecture/) for more context.  Users and clients cannot directly communicate with Applications RP. They instead communicate with UCP. UCP forwards relevant requests to Applications RP. 
+The RP has a datastore for storing Radius data, message queue for processing asynchronous requests, and a secret Store for storing sensitive information such as certificates. All these are configurable components and support multiple implementations.  
+
+### Architecture
+
+The RP consists of four types of resource providers for managing various types of resources in an application. `Applications.Core` resource provider manages core application resources such as application, environment, container and gateways. `Applications.Dapr` resource provider manages all dapr resources that are deployed as part of application. `Applications.Datastore` resource provider supports provisioning SQL database, Mongo DB and Redis Cache.
+`Applications.Messaging` resources provider manages queues such as Rabbit MQ.
+
+Applications RP has a key sub component `Recipe Engine` to execute `recipes`. 
+`Recipes` are Bicep or Terraform code supplied by user that is used to deploy infrastructure components on Azure and AWS. The Bicep recipes are fetched from OCI compliant registries. Terraform recipes are public modules and fetched from internet too. 
```

Comment:
Is it just on Azure or AWS? What about Kubernetes?

Reply:
I will update the text to include k8s resourecs as well.